VMR Stage2 failing to find NuGet Packages

The Stage 2 builds are failing to find NuGet Packages because no NuGet Packages be published as pipeline artifacts.

This [Failing build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2538107&view=results) (internal Microsoft link) is caused by https://github.com/dotnet/sdk/pull/43288:

`Copy Previous Build (CentOSStream9_Online_MsftSdk_x64 - NuGet Packages)` failed with below error message:

    ##[error]Error: Not found SourceFolder: /mnt/vss/_work/1/CentOSStream9_Online_MsftSdk_x64_Artifacts/packages'

Only 3 files be downloaded in `Download Previous Build (CentOSStream9_Online_MsftSdk_x64)` step:

    Minimatch patterns: [**/Private.SourceBuilt.Artifacts.*.tar.gz,**/dotnet-sdk-*.tar.gz,**/*.nupkg]
    Filtered 3 files from the Minimatch filters supplied.

--- a/eng/pipelines/templates/jobs/vmr-build.yml
+++ b/eng/pipelines/templates/jobs/vmr-build.yml
@@ -215,20 +215,22 @@ jobs:
           **/*.nupkg
         displayName: Download Previous Build (${{ reuseBuildArtifacts }})
 
-      - task: CopyFiles@2
-        displayName: Copy Previous Build (${{ reuseBuildArtifacts }} - Source Build artifacts)
-        inputs:
-          SourceFolder: $(Pipeline.Workspace)/${{ reuseBuildArtifacts }}_Artifacts/assets/Release
-          Contents: '*.tar.gz'
-          TargetFolder: $(sourcesPath)/prereqs/packages/archive/
+      - ${{ eq(parameters.buildSourceOnly, true) }}:
+        - task: CopyFiles@2
+          displayName: Copy Previous Build (${{ reuseBuildArtifacts }} - Source Build artifacts)
+          inputs:
+            SourceFolder: $(Pipeline.Workspace)/${{ reuseBuildArtifacts }}_Artifacts/assets/Release
+            Contents: '*.tar.gz'
+            TargetFolder: $(sourcesPath)/prereqs/packages/archive/
 
-      - task: CopyFiles@2
-        displayName: Copy Previous Build (${{ reuseBuildArtifacts }} - NuGet Packages)
-        inputs:
-          SourceFolder: $(Pipeline.Workspace)/${{ reuseBuildArtifacts }}_Artifacts/packages
-          Contents: '**/*.nupkg'
-          OverWrite: false
-          TargetFolder: $(sourcesPath)/artifacts/packages/
+      - ${{ else }}:
+        - task: CopyFiles@2
+          displayName: Copy Previous Build (${{ reuseBuildArtifacts }} - NuGet Packages)
+          inputs:
+            SourceFolder: $(Pipeline.Workspace)/${{ reuseBuildArtifacts }}_Artifacts/packages
+            Contents: '**/*.nupkg'
+            OverWrite: false
+            TargetFolder: $(sourcesPath)/artifacts/packages/
 
   - ${{ if eq(parameters.withPreviousSDK, 'true') }}:
     - script: |

